monitoring/src/monitoring_service/service.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import time
+from typing import Callable, List
+
+import structlog
+from web3 import Web3
+
+from monitoring_service.blockchain import BlockchainListener
+from monitoring_service.constants import DEFAULT_REQUIRED_CONFIRMATIONS
+from monitoring_service.database import Database
+from monitoring_service.events import Event, ScheduledEvent
+from monitoring_service.handlers import HANDLERS, Context
+from monitoring_service.states import BlockchainState, MonitoringServiceState
+from raiden_contracts.contract_manager import ContractManager
+
+log = structlog.get_logger(__name__)
+
+
+def handle_event(event: Event, context: Context):
+    log.debug('Processing event', event_=event)
+    handler: Callable = HANDLERS[type(event)]
+    handler(event, context)
+
+
+class MonitoringService:
+    def __init__(
+        self,
+        web3: Web3,
+        contract_manager: ContractManager,
+        private_key: str,
+        registry_address: str,
+        monitor_contract_address: str,
+        sync_start_block: int = 0,
+        required_confirmations: int = DEFAULT_REQUIRED_CONFIRMATIONS,
+        poll_interval: int = 1,
+    ):
+        self.web3 = web3
+        self.contract_manager = contract_manager
+        self.private_key = private_key
+        self.required_confirmations = required_confirmations
+        self.poll_interval = poll_interval
+
+        chain_state = BlockchainState(
+            token_network_registry_address=registry_address,
+            monitor_contract_address=monitor_contract_address,
+            latest_known_block=sync_start_block,
+        )
+        self.ms_state = MonitoringServiceState(
+            blockchain_state=chain_state,
+        )
+
+        # TODO: tie database to chain id
+        self.database = Database()
+        self.scheduled_events: List[ScheduledEvent] = list()
+
+        self.bcl = BlockchainListener(
+            web3=self.web3,
+            contract_manager=contract_manager,
+        )
+
+        self.context = Context(
+            ms_state=self.ms_state,
+            db=self.database,
+            scheduled_events=self.scheduled_events,
+            w3=self.web3,
+            contract_manager=contract_manager,
+            last_known_block=0,
+        )
+
+    def start(self, wait_function: Callable = time.sleep):
+        while True:
+            last_block = self.web3.eth.blockNumber - self.required_confirmations
```

Comment:
We'll miss previously opened channels if we only start here, won't we? We'll have to touch this anyway when loading/saving state between restart, again.

Reply:
Atm yes, but once the `BlockchainState` gets persisted (can one write that?) it should work just fine.